src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AvailableLocationsService.kt

```diff
@@ -80,14 +76,50 @@ class AvailableLocationsService(
     }
 
     return AvailableLocationsResponse(
-      availableLocations
-        .sortedWith(compareBy({ it.startTime }, { it.name }))
-        .distinctBy { it.startTime }
-        .also { log.info("AVAILABLE LOCATIONS: found ${it.size} available locations matching request $request") },
+      availableLocations.build().also { log.info("AVAILABLE LOCATIONS: found ${it.size} available locations matching request $request") },
     )
   }
 
-  private fun AvailableLocationsRequest.fallsWithinSlotTime(time: LocalTime) = timeSlots!!.any { slot -> slot.isTimeInSlot(time) }
+  private fun availableLocations(init: AvailableLocationsBuilder.() -> Unit): AvailableLocationsBuilder {
+    val availableLocations = AvailableLocationsBuilder()
+    availableLocations.init()
+    return availableLocations
+  }
+
+  private class AvailableLocationsBuilder {
+    private val probation = mutableSetOf<AvailableLocation>()
+    private val court = mutableSetOf<AvailableLocation>()
+    private val shared = mutableSetOf<AvailableLocation>()
+
+    fun add(availableLocation: AvailableLocation) {
+      // TODO handle probation code, court code, shared and scheduled types specifically
+      when (availableLocation.usage) {
+        LocationUsage.PROBATION -> probation.add(availableLocation)
+        LocationUsage.COURT -> court.add(availableLocation)
+        else -> shared.add(availableLocation)
+      }
+    }
+
+    fun build() = run {
+      if (probation.isNotEmpty() && court.isNotEmpty()) {
```

Comment:
This is where the logic for sorting the available locations will live.  It is a WIP